tests/ui/attributes/nonterminal-expansion.rs

```diff
@@ -15,5 +14,6 @@ macro_rules! n {
 }
 
 pass_nonterminal!(n!());
+//~^ ERROR expected one of `(`, `::`, or `=`, found `!`
```

Comment:
This error is surprisingly different from what you get when "inlining" the macro?
```
error: expected unsuffixed literal, found `!`
 --> src/main.rs:6:15
  |
6 | #[repr(align(n!()))]
  |               ^
```
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=979f349ee4be2e88b300d4f97f5e1db9

Reply:
`$expr` effectively has parentheses around it, so the "inlined" version would look more like `#[repr(align((n!())))]`.